src/app-bridge/app-bridge-state.ts

```diff
@@ -10,6 +11,17 @@ export type AppBridgeState = {
   theme: ThemeType;
   locale: LocaleCode;
   saleorApiUrl: string;
+  user?: {
+    /**
+     * Original permissions of the user that is using the app.
+     * *Not* the same permissions as the app itself.
+     *
+     * Can be used by app to check if user is authorized to perform
+     * domain specific actions
+     */
+    ownPermissions: Permission[];
+    email: string;
+  };
```

Comment:
@saleor/storefront-checkout FYI we will release such thing in next release

Thanks to that, you can easily verify if user using the app has permissions

Common use case:
A stuff member (non admin) can open the app to perform search index action in Algolia.
However, only admin (MANAGE_APPS permission) should be able to update Algolia tokens

Using `createProtectedHandler` you can provide require permission to the endpoint.
To show this on frontend (eg disabled form) you can check it easily with this feature now

Keep in mind that in such case there should be backend logic that validates token.
App (using apptoken) can update its own metadata. User token without "manage apps" can't.

So, to safely perform this operation you need to
1. Check `MANAGE_APPS` on the frontend (AppBridgeState), to give user feedback
2. Send form to API function, to check token again (token is validated and permission can't be passed)
3. API function will use AppToken to update its configuration, not user

cc @krzysztofwolski please verify what I said :)

Reply:
Well said :) 
Using the token from the app bridge allows us to perform actions on behalf of a particular Staff user:
- the Core will also check the permissions
- event logs (for example, timeline in the Order Events) will be able to display who performed the action and which application was used